File: src/main/java/com/server/ordering/domain/OrderFood.java
package com.server.ordering.domain;

import lombok.*;

import javax.persistence.*;

import static javax.persistence.FetchType.*;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
@RequiredArgsConstructor(access = PRIVATE)
public class OrderFood {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "order_food_id")
    private Long id;

    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "food_id")
    private Food food;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "order_id")
    private Order order;

    @NonNull
    private int price;
    @NonNull
    private int count;

    public void registerOrder(Order order) {
        this.order = order;
    }

    public static OrderFood createOrderFood(Food food, int price, int count) {
        return new OrderFood(food, price, count);
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/OrderFood.kt
package com.byeoru.ordering_server.domain

import javax.persistence.*

@Entity
class OrderFood(food: Food,
                price: Int,
                count: Int) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_food_id")
     var id: Long? = null
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "food_id")
     var food: Food = food
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
     var order: Order? = null
         protected set

     var price = price
         protected set

     var count = count
         protected set

    fun registerOrder(order: Order) {
        this.order = order
    }

    companion object {
        @JvmStatic
        fun createOrderFood(food: Food, price: Int, count: Int): OrderFood {
            return OrderFood(food, price, count)
        }
    }
}